---
title: "RNAi screen for piRNA pathway genes"
author: "Ming Wang"
date: "`r Sys.Date()`"
output:
  html_document:
    highlight: tango
    toc: yes
    toc_float:
      collapsed: no
    keep_md: true
  word_document:
    toc: yes
  pdf_document:
    toc: yes
---

```{r pkgs, echo = FALSE}
# load data
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(ggupset))
suppressPackageStartupMessages(library(ggplot2))
suppressPackageStartupMessages(library(knitr))
```


## I. Summary


+ 1. Highlight genes in all three screens

The following **11** genes were identified as piRNA pathway genes in all three screens.

```{r, echo = FALSE}
f1 <- "results/2.screen-377-144-87.txt"
df <- readr::read_delim(f1, "\t", col_types = readr::cols()) 

df %>%
  dplyr::filter(sum == 3) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::pull(Symbol)
```


+ 2. genes identified in at least **two** screens

The following **32** genes were identified in **two** screens. (the last one "NA" is gene "vas", the CG number was updated from "CG43081" to "CG46283" )

```{r, echo = FALSE}
df %>%
  dplyr::filter(sum == 2) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::pull(Symbol)
```


## II. Results

### 1. Group1: 74-49-87

Candidate genes:  
  
  - 74 genes, Czech Benjamin (Greg lab)    
  
  - 49 genes, Handler Dominik (Julius lab)    
  
  - 87 genes, Muerdter Felix (Greg lab)  

The positive genes reported in each screen.


```{r group1, echo = FALSE, fig.width = 6, fig.height = 4}

f1 <- "results/1.screen-74-80-87.txt"

df <- readr::read_delim(f1, "\t", col_types = readr::cols()) 

df1 <- df %>%
  tidyr::gather(group, status, 4:6) %>%
  dplyr::filter(status > 0) %>%
  group_by(FBID, CG, Symbol) %>%
  summarize(groups = list(group)) 

p1 <- df1 %>%
  ggplot(aes(groups)) + 
  geom_bar() +
  geom_text(aes(label = ..count..), stat = "count", vjust = -.3) +
  ggtitle("Genes Overlap between groups") +
  scale_x_upset()

print(p1)
```

Overlapped genes:

+ Overlap in all screens: 


```{r group1a, echo = FALSE}
df %>%
  dplyr::filter(sum == 3) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::mutate(
    Symbol = paste0("[", Symbol, "](http://flybase.org/reports/", FBID, ")"),
    FBID = paste0("[", FBID, "](http://flybase.org/reports/", FBID, ")")) %>%
  tibble::rowid_to_column("num") %>%
  kable()
```

+ Overlap in at least two screens

```{r group1b, echo = FALSE}
df %>%
  dplyr::filter(sum == 2) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::mutate(
    Symbol = paste0("[", Symbol, "](http://flybase.org/reports/", FBID, ")"),
    FBID = paste0("[", FBID, "](http://flybase.org/reports/", FBID, ")")) %>%
  tibble::rowid_to_column("num") %>%
  kable()
```



### 2. Group2: 377-144-87

Candidate genes:  
  
  - 377 genes, Czech Benjamin (Greg lab)    
  
  - 144 genes, Handler Dominik (Julius lab)    
  
  - 87 genes, Muerdter Felix (Greg lab)  

For Benjamin screen, instead use average Z-score of four TEs <= -2.0 (74 genes), at least Z-score <= -2.0 for **one** TE. (377 genes).  

For Dominik screen, change the criteria of staining from 2.0 (49 genes) to 1.0 (144 genes).


```{r group2, echo = FALSE, fig.width = 6, fig.height = 4}
# load data
library(readr)
library(dplyr)
library(ggupset)
library(ggplot2)

f1 <- "results/2.screen-377-144-87.txt"

df <- readr::read_delim(f1, "\t", col_types = readr::cols()) 

df1 <- df %>%
  tidyr::gather(group, status, 4:6) %>%
  dplyr::filter(status > 0) %>%
  group_by(FBID, CG, Symbol) %>%
  summarize(groups = list(group)) 

p1 <- df1 %>%
  ggplot(aes(groups)) + 
  geom_bar() +
  geom_text(aes(label = ..count..), stat = "count", vjust = -.3) +
  ggtitle("Genes Overlap between groups") +
  scale_x_upset()

print(p1)
```

+ Overlaped in all three screens


```{r group2a, echo = FALSE}
df %>%
  dplyr::filter(sum == 3) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::mutate(
    Symbol = paste0("[", Symbol, "](http://flybase.org/reports/", FBID, ")"),
    FBID = paste0("[", FBID, "](http://flybase.org/reports/", FBID, ")")) %>%
  tibble::rowid_to_column("num") %>%
  kable()
```


+ Overlaped, in at least **two** screens


```{r group2b, echo = FALSE}
df %>%
  dplyr::filter(sum == 2) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::mutate(
    Symbol = paste0("[", Symbol, "](http://flybase.org/reports/", FBID, ")"),
    FBID = paste0("[", FBID, "](http://flybase.org/reports/", FBID, ")")) %>%
  tibble::rowid_to_column("num") %>%
  kable()
```



### 3. Group2: 137-144-87

Candidate genes:  
  
  - 137 genes, Czech Benjamin (Greg lab)    
  
  - 144 genes, Handler Dominik (Julius lab)    
  
  - 87 genes, Muerdter Felix (Greg lab)  

For Benjamin screen, instead use average Z-score of four TEs <= -2.0 (74 genes), at least Z-score <= -2.0 for **two** TEs. (137 genes).  

For Dominik screen, change the criteria of staining from 2.0 (49 genes) to 1.0 (144 genes).


```{r group3, echo = FALSE, fig.width = 6, fig.height = 4}
# load data
library(readr)
library(dplyr)
library(ggupset)
library(ggplot2)

f1 <- "results/3.screen-137-144-87.txt"

df <- readr::read_delim(f1, "\t", col_types = readr::cols()) 

df1 <- df %>%
  tidyr::gather(group, status, 4:6) %>%
  dplyr::filter(status > 0) %>%
  group_by(FBID, CG, Symbol) %>%
  summarize(groups = list(group)) 

p1 <- df1 %>%
  ggplot(aes(groups)) + 
  geom_bar() +
  geom_text(aes(label = ..count..), stat = "count", vjust = -.3) +
  ggtitle("Genes Overlap between groups") +
  scale_x_upset()

print(p1)
```


+ Overlaped in all three screens


```{r group3a, echo = FALSE}
df %>%
  dplyr::filter(sum == 3) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::mutate(
    Symbol = paste0("[", Symbol, "](http://flybase.org/reports/", FBID, ")"),
    FBID = paste0("[", FBID, "](http://flybase.org/reports/", FBID, ")")) %>%
  tibble::rowid_to_column("num") %>%
  kable()
```


+ Overlaped, in at least two screens


```{r group3b, echo = FALSE}
df %>%
  dplyr::filter(sum == 2) %>%
  dplyr::select(1:3) %>%
  dplyr::arrange(Symbol) %>%
  dplyr::mutate(
    Symbol = paste0("[", Symbol, "](http://flybase.org/reports/", FBID, ")"),
    FBID = paste0("[", FBID, "](http://flybase.org/reports/", FBID, ")")) %>%
  tibble::rowid_to_column("num") %>%
  kable()
```


## III. Materials


### 1. Czech paper (Greg lab)


**Where**

  - Ovary, Germline specific genes

**How**

  - TE derepression upon knockdown of given genes.

**What** 
  
  - knock down genes by dsRNA;   
  - Check the expression of TEs (HetA, TAHRE, blood, burdock); (derepression); (LINE:HeT-A, TAHRE; LTR: blood, burdock), armi=positive, white=negative;
  - Filter by Z-score;  

**Results**
  
  - A total of 8171 genes were screened. [table S1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx).
  - 74 genes (Average Z-score <= -2.0) ;   
  - 216 genes (Average Z-score <= -1.5) ;   
  - 137 genes (Z-score <= -2.0, at least in two TEs) ;    
  - 377 genes (Z-score <= -2.0, at least in one TE) ;    

See [table S1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx) for details.


### 2. Handler paper (Julius lab)


**Where**

  - Ovary; somatic piRNA pathway

**What**
  
  - reporter: gypsy-LTR + beta-gal (controled by flam derived piRNAs)    
  - X-gal staining

**How**
  
  - The expression of flam was controled by piRNA pathway. (repression, no staining, white)    
  - The repression of flam was disrupted by specific gene knockdown (staining, blue)

**Results**
  
  - 49 genes, (staining >=2) ;      
  - 80 genes, (staining >= 1.5) ;   
  - 144 genes, (staining >= 1) ;
  
**Notes**

  - 7257 genes expressed in OSC (RPKM >= 1).    
  - VDRC 6818 genes, (2 crosses per gene).


See [tabld S1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513003365-mmc2.xlsx) for details.


### 3. Muerdter paper (Greg lab)


**Where**

  - OSS celline


**What**   
  
  - Check gypsy, ZAM, gypsy3 expression by qPCR ;    
  - TE derepression upon knockdown (KD) of any given genes


**How**  
  
  - genes were knockdown by dsRNA


**Results**
  
  - 87 genes, validated     
  - 52 genes, development defect      
  - 149 genes, non validated    

**Notes**

  - H3K9me3 decreased, piRNA levels not changed.

see [table S3](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc3.xlsx) for details.



## Iv. Code


```{r, echo = FALSE}
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(AnnotationDbi))
suppressPackageStartupMessages(library(org.Dm.eg.db))
```


### 1. Czech (Greg lab)

```{r p1}
## Benjamin Czech, 2013 (Greg lab)
## table S1: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx
## 
## Table S1. Raw Data for Primary Screen, Related to Figure 2. Shown are transformant IDs (of VDRC stock center; TID), library (Lib), viability of fly stock, and chromosome insertion (Chr) of the dsRNA for all 8,171 RNAi lines used. Gene IDs and synonyms are shown along with raw Z scores for HeTA, TAHRE, blood, and burdock.
##
## derepression threshold, Z-score
## genes; Z-score
## 74    -2.0
## 216   -1.5
## 137   -2.0 in at least two TE
## 377   -2.0 in at least one TE
fileURL <- "https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx"
fileTableS1  <- "data/1.Ben_2013_Greg_lab/TableS1.xlsx"
# download.file(fileURL, fileTableS1, method = "curl")

## including NA values (n.d.)
## 8171 records
df <- readxl::read_xlsx(fileTableS1, na = "n.d.") %>%
  dplyr::mutate(HeTA = as.numeric(HeTA),
                TAHRE = as.numeric(TAHRE),
                blood = as.numeric(blood),
                burdock = as.numeric(burdock))

## 8133 records
df$mean <- df[, 5:8] %>%
  rowMeans(na.rm = TRUE)
df <- dplyr::arrange(df, mean)

## df <- df[complete.cases(df), ]
df[is.na(df)] <- 0 

## threshold: -2.0 for each condition
df$status <- purrr::map(df[, 5:8], function(x) x <= -2.0) %>% 
  as.data.frame() %>%
  rowSums()

## threshold: -2.0, -1.5
sum(df$mean <= -2.0) # 74
sum(df$mean <= -1.5) # 216
sum(df$status >= 2)  # 137, at least in two TE
sum(df$status >= 1)  # 377, at least in two TE

## total
df1 <- df %>%
  dplyr::filter(status >= 1) %>%
  dplyr::mutate(FBID = mapIds(org.Dm.eg.db, keys = Gene, keytype = "FLYBASECG", column = "FLYBASE"),
                Symbol = mapIds(org.Dm.eg.db, keys = Gene, keytype = "FLYBASECG", column = "SYMBOL"),
                CG = Gene) %>%
  dplyr::select(FBID, CG, Symbol, mean, status)

## save to table
file1 <- "data/1.Ben_2013_Greg_lab/01_screen_Ben_377.txt"
readr::write_delim(df1, file1, delim = "\t", na = "n.d.", col_names = TRUE) 

## 74 genes
df2 <- df1 %>%
  dplyr::filter(mean <= -2.0)

file2 <- "data/1.Ben_2013_Greg_lab/01_screen_Ben_74.txt"
readr::write_delim(df2, file2, delim = "\t", na = "n.d.", col_names = TRUE) 


## 137 genes
df3 <- df1 %>%
  dplyr::filter(mean <= -1.5)

file3 <- "data/1.Ben_2013_Greg_lab/01_screen_Ben_137.txt"
readr::write_delim(df3, file3, delim = "\t", na = "n.d.", col_names = TRUE) 

```



### 2. Handler (Julius lab)

```{r}
## Handler, 2013 (Julis lab)
## table S1: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513003365-mmc2.xlsx
##
## Table S1. Primary and Verified Results of the Screen.
##
## genes (staining)
## 49 (>=2)
## 80 (>=1.5)
## 144 (>=1)
fileURL <- "https://ars.els-cdn.com/content/image/1-s2.0-S1097276513003365-mmc2.xlsx"
fileTableS1 <- "data/2.Handler_2013_Julius_lab/TableS1.xlsx"
# download.file(fileURL, fileTableS1, method = "curl")

## including blanks
df <- readxl::read_xlsx(fileTableS1, trim_ws = TRUE, skip = 1) %>% 
  dplyr::select(c(1, 2, 4, 6, 7, 9:14)) 

names(df) <- c("Symbol", "CG", "RPKM", "staining", "morphology",
               "VDRC_1", "staining_1", "morphology_1", 
               "VDRC_2", "staining_2", "morphology_2")
## staining
df_staining <- data.frame(
  value = c(0, 1, 1.5, 2, 2.5, 3),
  status = c("no staining", "weak staining",
             "intermediate-weak staining", "intermediate staining",
             "strong-intermediate staining", "strong staining"),
  stringsAsFactors = FALSE
)

df <- df %>% 
  dplyr::mutate(status = plyr::mapvalues(staining, 
                                         from = df_staining$value,
                                         to = df_staining$status))
# df <- df[complete.cases(df), ]
df[is.na(df)] <- 0

sum(df$staining >= 2)   # 49
sum(df$staining >= 1.5) # 80
sum(df$staining >= 1)   # 144

## pick 144 genes
df <- df %>%
  dplyr::filter(staining >= 1)

df1 <- df %>%
  dplyr::mutate(
    FBID = mapIds(org.Dm.eg.db, keys = CG, keytype = "FLYBASECG", column = "FLYBASE"),
    Symbol = mapIds(org.Dm.eg.db, keys = CG, keytype = "FLYBASECG", column = "SYMBOL")) %>%
  dplyr::select(FBID, CG, Symbol, staining, status)

## save to table
file1 <- "data/2.Handler_2013_Julius_lab/02_screen_Handler_144.txt"
readr::write_delim(df1, file1, delim = "\t", na = "n.d.", col_names = TRUE)

## 80 genes
df2 <- df1 %>%
  dplyr::filter(staining >= 1.5)
##
file2 <- "data/2.Handler_2013_Julius_lab/02_screen_Handler_80.txt"
readr::write_delim(df2, file2, delim = "\t", na = "n.d.", col_names = TRUE)


```


### 3. Muerdter (Greg lab)

```{r}
## Muerdter, 2013 (Greg lab)
## table S1: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc2.xlsx
## table S3: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc3.xlsx
## Table S1. Primary Screen Results, Related to Figure 1.
## Table S3. Validation Screen Results, Related to Figure 2.
##
## genes (validated)
## 86 (va)
## 
fileURL <- "https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc3.xlsx"
fileTableS3 <- "data/3.Muertder_2013_Greg_lab/TableS3.xlsx"
# download.file(fileURL, fileTableS3, method = "curl")

## parsing data
df <- readxl::read_xlsx(fileTableS3, trim_ws = TRUE)
colnames(df) <- c("FBID", "TRANSID", "Symbol", "Fertility",
                  "fc_gypsy", "fc_gypsy3", "fc_ZAM", "status")

## 87 validated genes
length(
  df %>% 
    dplyr::filter(status %in% c("va")) %>%
    dplyr::pull("FBID") %>%
    unique())

## 52 dev defect
length(
  df %>% 
    dplyr::filter(status %in% c("dd")) %>%
    dplyr::pull("FBID") %>%
    unique())

## only va genes
df1 <- df %>% 
  dplyr::filter(status %in% c("va")) %>%
  dplyr::mutate(
    CG = mapIds(org.Dm.eg.db, keys = FBID, keytype = "FLYBASE", column = "FLYBASECG")
  ) %>%
  dplyr::select(FBID, CG, Symbol, status)


## save to table
file <- "data/3.Muertder_2013_Greg_lab/03_screen_Muerdter_87.txt"
readr::write_delim(df1, file, delim = "\t", na = "n.d.", col_names = TRUE)

```


## V. Reference

1.	Czech, B., Preall, J. B., McGinn, J. & Hannon, G. J. A Transcriptome-wide RNAi Screen in the Drosophila Ovary Reveals Factors of the Germline piRNA Pathway. Mol. Cell 50, 749–761 (2013). DOI: [10.1016/j.molcel.2013.04.007](https://doi.org/10.1016/j.molcel.2013.04.007)

2.	Handler, D. et al. The Genetic Makeup of the Drosophila piRNA Pathway. Mol. Cell 50, 762–777 (2013). DOI: [10.1016/j.molcel.2013.04.031](https://doi.org/10.1016/j.molcel.2013.04.031)    
3.	Muerdter, F. et al. A Genome-wide RNAi Screen Draws a Genetic Framework for Transposon Control and Primary piRNA Biogenesis in Drosophila. Mol. Cell 50, 736–748 (2013). DOI: [10.1016/j.molcel.2013.04.006](https://doi.org/10.1016/j.molcel.2013.04.006)    



