---
title: "RNAi screen for piRNA pathway genes"
author: "Ming Wang"
date: "`r Sys.Date()`"
output:
  html_document:
    highlight: tango
    toc: yes
    toc_float:
      collapsed: no
    keep_md: true
  word_document:
    toc: yes
  pdf_document:
    toc: yes
---

```{r pkgs, echo = FALSE}
# load data
library(readr, quietly = T)
library(dplyr, quietly = T)
library(ggplot2, quietly = T)
library(ggupset, quietly = T)
library(knitr, quietly = T)
```

## README

Here are the hits of RNAi screen in fruitfly, by Greg and Julius lab in 2013, published on Mol Cell.

+ Ben (Greg lab)

  - 1. mean value <= -2.0 (n=74)
  
  - 2. mean value <= -1.5 (n=216)
  
  - 3. at least in two TEs, <= -2.0 (n=137)
  
  - 4. at least in one TE,  <= -2.0 (n=377)  

+ Handler (Julius lab) 

  - 1. staining >= 2.0 (n=49)
  
  - 2. staining >= 1.5 (n=80)
  
  - 3. staining >= 1.0 (n=144)

+ Muerdter (Greg lab) 

  - 1. va, validated genes (n=87)
  
  - 2. dd, development defect, (n=52)
  
  - 3. nv, non validated, (n=149)
  
## 1. gene list (137 + 144 + 87)

```{r, echo = FALSE}
f1  <- "data/fixed_list/screen_hits.137_144_87.csv"
df1 <- readr::read_csv(f1, show_col_types = F)
print(table(df1$sum))
```

A total of **328** genes. **9** genes identified in all three screen; **22** genes were identified in two screen; **297** were identified in only one screen.

# <a href="" ></a>

```{r group1a, results="asis", echo = FALSE}
df1 %>%
  dplyr::mutate(
    gene_id = paste0(
      "<a href=http://flybase.org/reports/", gene_id, " target='_blank'>", gene_id, "</a>")
    # gene_id = paste0("[", gene_id, "](http://flybase.org/reports/", gene_id, ")")
  ) %>% 
  DT::datatable(
    extensions = 'Buttons',
    rownames = FALSE,
    escape   = FALSE, # show html code
    options  = list(
      dom        = 'Bftpl',
      pageLength = 10,
      scrollX    = TRUE,
      extensions = 'Buttons',
      buttons    = list('copy',
                        list(extend  = "collection",
                             buttons = "excel",
                             text    = "Download"))
    )
  )
```


```{r group1b, echo = FALSE, eval = F}
df1b <- list(
  ben_137     = dplyr::filter(df1, ben_137 > 0) %>% pull("gene_id"),
  handler_144 = dplyr::filter(df1, handler_144 > 0) %>% pull("gene_id"),
  muerdter_87 = dplyr::filter(df1, muerdter_87 > 0) %>% pull("gene_id")
)
hisequtils::overlap(df1b, labels = c("ben_137", "handler_144", "muerdter_87"))
```

```{r group1c, echo = F}
df1b <- df1 %>%
  tidyr::pivot_longer(4:6, names_to = "group", values_to = "status") %>%
  dplyr::filter(status > 0) %>%
  group_by(gene_id, cg_id, symbol) %>%
  summarize(groups = list(group), .groups = "drop") 

p1 <- df1b %>%
  ggplot(aes(groups)) + 
  geom_bar() +
  geom_text(aes(label = ..count..), stat = "count", vjust = -.3) +
  ggtitle("Genes Overlap between groups") +
  scale_x_upset() + 
  theme_bw()
print(p1)
```


## 2. Further process

Among 328 genes from above screen hits.

+ 2.1  **52** genes were not tested 

  - 2, already tested: `tested`
  
  - 17, piRNA biogenesis related genes: `piRNA_biogenesis`
  
  - 33, ribosomal genes: `ribosomal`
  
+ 2.2 **60** genes were failed for further analysis

  - 3, no F1: `no_F1`
  
  - 57, not available at the time of test: `n.a.`
  
```{r table2 , echo = F}
f2  <- "data/fixed_list/screen_hits.137_144_87.not_test.csv"
df2 <- readr::read_csv(f2, show_col_types = F)
df2$group[is.na(df2$group)] <- "n.a."
# show table
df2 %>%
  dplyr::mutate(
    gene_id = paste0(
      "<a href=http://flybase.org/reports/", gene_id, " target='_blank'>", gene_id, "</a>")
    # gene_id = paste0("[", gene_id, "](http://flybase.org/reports/", gene_id, ")")
  ) %>% 
  DT::datatable(
    extensions = 'Buttons',
    rownames = T,
    escape   = FALSE, # show html code
    options  = list(
      dom        = 'Bftpl',
      pageLength = 10,
      scrollX    = TRUE,
      extensions = 'Buttons',
      buttons    = list('copy',
                        list(extend  = "collection",
                             buttons = "excel",
                             text    = "Download"))
    )
  )
```


## 3. **216** genes were used for further analysis


```{r table3, echo = F}
df3 <- df1 %>%
  dplyr::filter(! gene_id %in% df2$gene_id) %>%
    DT::datatable(
    extensions = 'Buttons',
    rownames = T,
    escape   = FALSE, # show html code
    options  = list(
      dom        = 'Bftpl',
      pageLength = 10,
      scrollX    = TRUE,
      extensions = 'Buttons',
      buttons    = list('copy',
                        list(extend  = "collection",
                             buttons = "excel",
                             text    = "Download"))
    )
  )
```



## III. Materials


### 1. Czech paper (Greg lab)


**Where**

  - Ovary, Germline specific genes

**How**

  - TE derepression upon knockdown of given genes.

**What** 
  
  - knock down genes by dsRNA;   
  - Check the expression of TEs (HetA, TAHRE, blood, burdock); (derepression); (LINE:HeT-A, TAHRE; LTR: blood, burdock), armi=positive, white=negative;
  - Filter by Z-score;  

**Results**
  
  - A total of 8171 genes were screened. [table S1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx).
  - 74 genes (Average Z-score <= -2.0) ;   
  - 216 genes (Average Z-score <= -1.5) ;   
  - 137 genes (Z-score <= -2.0, at least in two TEs) ;    
  - 377 genes (Z-score <= -2.0, at least in one TE) ;    

See [table S1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx) for details.


### 2. Handler paper (Julius lab)


**Where**

  - Ovary; somatic piRNA pathway

**What**
  
  - reporter: gypsy-LTR + beta-gal (controled by flam derived piRNAs)    
  - X-gal staining

**How**
  
  - The expression of flam was controled by piRNA pathway. (repression, no staining, white)    
  - The repression of flam was disrupted by specific gene knockdown (staining, blue)

**Results**
  
  - 49 genes, (staining >=2) ;      
  - 80 genes, (staining >= 1.5) ;   
  - 144 genes, (staining >= 1) ;
  
**Notes**

  - 7257 genes expressed in OSC (RPKM >= 1).    
  - VDRC 6818 genes, (2 crosses per gene).


See [tabld S1](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513003365-mmc2.xlsx) for details.


### 3. Muerdter paper (Greg lab)


**Where**

  - OSS celline


**What**   
  
  - Check gypsy, ZAM, gypsy3 expression by qPCR ;    
  - TE derepression upon knockdown (KD) of any given genes


**How**  
  
  - genes were knockdown by dsRNA


**Results**
  
  - 87 genes, validated     
  - 52 genes, development defect      
  - 149 genes, non validated    

**Notes**

  - H3K9me3 decreased, piRNA levels not changed.

see [table S3](https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc3.xlsx) for details.



## Iv. Code


```{r, echo = FALSE}
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(AnnotationDbi))
suppressPackageStartupMessages(library(org.Dm.eg.db))
```


### 1. Czech (Greg lab)

```{r p1}
#------------------------------------------------------------------------------#
# 1. Greg lab, Ben 2013
#------------------------------------------------------------------------------#
#
# Benjamin Czech, 2013 (Greg lab)
# table S1: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx
# 
# Table S1. Raw Data for Primary Screen, Related to Figure 2. Shown are 
# transformant IDs (of VDRC stock center; TID), library (Lib), viability of fly
# stock, and chromosome insertion (Chr) of the dsRNA for all 8,171 RNAi lines 
# used. Gene IDs and synonyms are shown along with raw Z scores for HeTA, TAHRE,
# blood, and burdock.
#
# derepression threshold, Z-score
# genes; Z-score
# 74    -2.0
# 216   -1.5
# 137   -2.0 in at least two TE
# 377   -2.0 in at least one TE
fileURL <- "https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002888-mmc2.xlsx"
fileTableS1  <- "data/raw_list/1.0.Greg_lab_Ben_2013.TableS1.xlsx"
# download.file(fileURL, fileTableS1, method = "curl")
#
# including NA values (n.d.)
# 8171 records, 11 columns
df1 <- readxl::read_xlsx(fileTableS1, na = "n.d.") %>%
  dplyr::mutate(across(all_of(c("HeTA", "TAHRE", "blood", "burdock")), as.numeric))
print(dim(df1))
# mean values
df1 <- df1 %>% 
  dplyr::mutate(mean_col = rowMeans(dplyr::select(., HeTA:burdock), na.rm = TRUE)) %>%
  dplyr::arrange(mean_col)
# fix n.a. values
df1[is.na(df1)] <- 0
df1$status <- rowSums(dplyr::select(df1, HeTA:burdock) <= -2.0)
#-------------------------#
# threshold: -2.0, -1.5
#-------------------------#
## 1. mean value <= -2.0 (n=74)
## 2. mean value <= -1.5 (n=216)
## 3. at least in two TEs, <= -2.0 (n=137)
## 4. at least in one TE,  <= -2.0 (n=377)
df1 <- df1 %>%
  dplyr::mutate(
    tag1 = as.numeric(mean_col <= -2),
    tag2 = as.numeric(mean_col <= -1.5),
    tag3 = as.numeric(status   >= 2),
    tag4 = as.numeric(status   >= 1)
  )
colSums(dplyr::select(df1, contains("tag")))
# df1 <- df1[df1$status > 0, ]
#-------------------------#
# save each table to files
#-------------------------#
## tag1: mean_value <= -2.0 (n=74)
f1a  <- "data/raw_list/1.1.Greg_lab_Ben_2013_74.txt"
df1a <- dplyr::filter(df1, tag1 > 0)
readr::write_delim(df1a, f1a, na = "n.d.", col_names = TRUE)
## tag2: mean_value <= -1.5 (n=216)
f1a  <- "data/raw_list/1.2.Greg_lab_Ben_2013_216.txt"
df1a <- dplyr::filter(df1, tag2 > 0)
readr::write_delim(df1a, f1a, na = "n.d.", col_names = TRUE)
## tag3: at least 2 TEs, <= -2.0 (n=137)
f1a  <- "data/raw_list/1.3.Greg_lab_Ben_2013_137.txt"
df1a <- dplyr::filter(df1, tag3 > 0)
readr::write_delim(df1a, f1a, na = "n.d.", col_names = TRUE)
## tag4: at least 1 TEs, <= -2.0 (n=377)
f1a  <- "data/raw_list/1.4.Greg_lab_Ben_2013_377.txt"
df1a <- dplyr::filter(df1, tag4 > 0)
readr::write_delim(df1a, f1a, na = "n.d.", col_names = TRUE)
```



### 2. Handler (Julius lab)

```{r}
#------------------------------------------------------------------------------#
# 2. Julius lab, Handler 2013
#------------------------------------------------------------------------------#
# Handler, 2013 (Julis lab)
# table S1: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513003365-mmc2.xlsx
#
# Table S1. Primary and Verified Results of the Screen.
#
# genes (staining)
# 49 (>=2)
# 80 (>=1.5)
# 144 (>=1)
fileURL <- "https://ars.els-cdn.com/content/image/1-s2.0-S1097276513003365-mmc2.xlsx"
fileTableS1 <- "data/raw_list/2.0.Julius_lab_Handler_2013.TableS1.xlsx"
# download.file(fileURL, fileTableS1, method = "curl")
#
# loading data from Excel file (13948 rows, 11 columns)
df2 <- readxl::read_xlsx(fileTableS1, trim_ws = TRUE, skip = 1) %>% 
  dplyr::select(c(1, 2, 4, 6, 7, 9:14)) 
colnames(df2) <- c(
  "Symbol", "CG", "RPKM", "staining", "morphology", "VDRC_1", "staining_1", "morphology_1", 
  "VDRC_2", "staining_2", "morphology_2"
)
# fix NA values in staining
df2$staining[is.na(df2$staining)] <- 0
print(dim(df2))
# Assign a new column "status" based on the "staining" column
df2 <- df2 %>%
  mutate(
    status = case_when(
      staining == 0   ~ "no staining",
      staining == 1   ~ "weak staining",
      staining == 1.5 ~ "intermediate-weak staining",
      staining == 2   ~ "intermediate staining",
      staining == 2.5 ~ "strong-intermediate staining",
      staining == 3   ~ "strong staining",
      TRUE ~ "no staining"  # Handle any other values (optional)
    )
  )
# filter records by staining
#-------------------------#
# threshold: staining
#-------------------------#
## 1. >= 2.0 (n=49)
## 2. >= 1.5 (n=80)
## 3. >= 1.0 (n=144)
df2 <- dplyr::mutate(
  df2, 
  tag1 = as.numeric(staining >= 2),
  tag2 = as.numeric(staining >= 1.5),
  tag3 = as.numeric(staining >= 1)
)
colSums(dplyr::select(df2, contains("tag")))
#-------------------------#
# save each table to files
#-------------------------#
## tag1: staining >= 2.0 (n=49)
f2a  <- "data/raw_list/2.1.Julius_lab_Handler_2013_49.txt"
df2a <- dplyr::filter(df2, tag1 > 0)
readr::write_delim(df2a, f2a, na = "n.d.", col_names = TRUE)
## tag1: staining >= 1.5 (n=80)
f2a  <- "data/raw_list/2.2.Julius_lab_Handler_2013_80.txt"
df2a <- dplyr::filter(df2, tag2 > 0)
readr::write_delim(df2a, f2a, na = "n.d.", col_names = TRUE)
## tag1: staining >= 1.0 (n=144)
f2a  <- "data/raw_list/2.3.Julius_lab_Handler_2013_144.txt"
df2a <- dplyr::filter(df2, tag3 > 0)
readr::write_delim(df2a, f2a, na = "n.d.", col_names = TRUE)
```


### 3. Muerdter (Greg lab)

```{r}
#------------------------------------------------------------------------------#
# 3. Greg lab, Muerdter 2013
#------------------------------------------------------------------------------#
# Muerdter, 2013 (Greg lab)
# table S1: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc2.xlsx
# table S3: https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc3.xlsx
# Table S1. Primary Screen Results, Related to Figure 1.
# Table S3. Validation Screen Results, Related to Figure 2.
#
# genes (validated)
# 86 (va)
# 
fileURL <- "https://ars.els-cdn.com/content/image/1-s2.0-S1097276513002876-mmc3.xlsx"
fileTableS3 <- "data/raw_list/3.0.Greg_lab_Muertder_2013.TableS3.xlsx"
# download.file(fileURL, fileTableS3, method = "curl")
#
# loading data (341 rows, 8 columns)
df3 <- readxl::read_xlsx(fileTableS3, trim_ws = TRUE)
colnames(df3) <- c(
  "FBID", "TRANSID", "Symbol", "Fertility", "fc_gypsy", "fc_gypsy3", "fc_ZAM", "status"
)
print(dim(df3))
# unique gene_id (292)
df3 <- dplyr::select(df3, FBID, Symbol, status) %>%
  unique
print(dim(df3))
print(table(df3$status))
# va, validated genes (n=87)
# dd, development defect, (n=52)
# nv, non validated, (n=149)
#-------------------------#
# save each table to files
#-------------------------#
## tag1: va, validated (n=87)
f3a  <- "data/raw_list/3.1.Greg_lab_Muerdter_2013_87.txt"
df3a <- dplyr::filter(df3, status == "va")
readr::write_delim(df3a, f3a, na = "n.d.", col_names = TRUE)
## tag2: dd, development defect (n=52)
f3a  <- "data/raw_list/3.2.Greg_lab_Muerdter_2013_52.txt"
df3a <- dplyr::filter(df3, status == "dd")
readr::write_delim(df3a, f3a, na = "n.d.", col_names = TRUE)
## tag3: nv, non validated (n=149)
f3a  <- "data/raw_list/3.3.Greg_lab_Muerdter_2013_149.txt"
df3a <- dplyr::filter(df3, status == "nv")
readr::write_delim(df3a, f3a, na = "n.d.", col_names = TRUE)
```


## V. Reference

1.	Czech, B., Preall, J. B., McGinn, J. & Hannon, G. J. A Transcriptome-wide RNAi Screen in the Drosophila Ovary Reveals Factors of the Germline piRNA Pathway. Mol. Cell 50, 749–761 (2013). DOI: [10.1016/j.molcel.2013.04.007](https://doi.org/10.1016/j.molcel.2013.04.007)

2.	Handler, D. et al. The Genetic Makeup of the Drosophila piRNA Pathway. Mol. Cell 50, 762–777 (2013). DOI: [10.1016/j.molcel.2013.04.031](https://doi.org/10.1016/j.molcel.2013.04.031)    
3.	Muerdter, F. et al. A Genome-wide RNAi Screen Draws a Genetic Framework for Transposon Control and Primary piRNA Biogenesis in Drosophila. Mol. Cell 50, 736–748 (2013). DOI: [10.1016/j.molcel.2013.04.006](https://doi.org/10.1016/j.molcel.2013.04.006)    



